src/commands/downloadDevContainer.ts

```diff
@@ -0,0 +1,63 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { WorkspaceFolder } from "vscode";
+import { IActionContext } from "vscode-azureextensionui";
+import { getDevContainerName } from '../downloadAzureProject/setupProjectFolder';
+import { ext } from '../extensionVariables';
+import { localize } from "../localize";
+import { ProductionSlotTreeItem } from '../tree/ProductionSlotTreeItem';
+import { SlotTreeItemBase } from '../tree/SlotTreeItemBase';
+import { requestUtils } from '../utils/requestUtils';
+import { selectWorkspaceFolder } from '../utils/workspace';
+import { tryGetFunctionProjectRoot } from './createNewProject/verifyIsProject';
+
+
+export async function downloadDevContainer(context: IActionContext, node?: SlotTreeItemBase): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<SlotTreeItemBase>(ProductionSlotTreeItem.contextValue, context);
+    }
+
+    const language: string = await node.getApplicationLanguage();
+    const devContainerName = getDevContainerName(language);
+    if (devContainerName && node.site.reserved) {
+        const message: string = localize('selectDevContainer', 'Select the destination file for the dev container folder.');
+        let devContainerFolderPath: string = await getDevContainerFolder(context, message);
+
+        if (devContainerFolderPath.substring(devContainerFolderPath.length - 13) != '.devcontainer') {
+            devContainerFolderPath = path.join(devContainerFolderPath, '.devcontainer');
+        }
+        await requestUtils.downloadFile(
```

Comment:
make the two file downloads parallel.

Reply:
In fact, I think Chloe or Fadi have this already in one of their PRs?? if so, can we reuse once they push in?